orchestrator/kafka/root.go

```diff
@@ -36,19 +36,25 @@ type Dependencies struct {
 
 // New creates a new Kafka configurator.
 func New(r *reporter.Reporter, config Configuration, dependencies Dependencies) (*Component, error) {
-	kafkaOpts, err := kafka.NewConfig(r, config.Configuration)
-	if err != nil {
-		return nil, err
-	}
-
 	c := Component{
 		r:      r,
 		d:      dependencies,
 		config: config,
+	}
+
+	if !config.ManageTopic {
+		r.Info().Msg("Kafka topic management disabled, skipping Kafka initialization")
+		return &c, nil
```

Comment:
I would suggest to just return `nil, nil` early and handle the nil case in `Start()` and `Stop()`.